portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/** A comment in comment section. */
+public final class Comment {
+
+  private final long id;
+  private final String input;
+  private final long timestamp;
+
+  public Comment(long id, String input, long timestamp) {
+    this.id = id;
+    this.input = input;
+    this.timestamp = timestamp;
+  }
```

Comment:
What do you think about creating a static method called fromEntity(Entity entity) that allows you to create a comment from an Entity object?

See below for the cool things you can do with this method.

Reply:
#### Comment.java
```java
public Comment(Entity entity){
    this.id = entity.getKey().getId();
    this.input = (String) entity.getProperty("input");
    this.timestamp = (long) entity.getProperty("timestamp");
 }
```
#### DataServlet.java
```javascript
for (Entity entity : results.asIterable()) {
    Comment comment = new Comment(entity);
    comments.add(comment);
}
```

Great suggestion! Bouncing off of what you're saying, what if I replaced the constructor to handle the entity directly like above? It would reduce a lot of redundant code in the Comment and DataServlet classes.

However, I'm not sure how it would tie into the syntax you've shown above.

Reply:
Typically, we like to have one constructor with parameters representing every attribute of the class. This is a code health best practice -- it makes your code more resilient to refactoring (what if we stop using entity and start using another class?).

Creating a static method instead of overloading the constructor allows you to take advantage of some cool java language features (the lambdas, streams, and method references).

the static method could look like this:

```java
public static Comment fromEntity(Entity entity) {
  return new Comment(entity.getProperty("input")....);
}
```

You could also use a forEach here:

as static method:
```java
results.asIterable().forEach(result -> comments.add(Comment.fromEntity(result));
```
or as a constructor

as overloaded constructor:
```java
results.asIterable().forEach(result -> comments.add(new Comment(entity));
```